multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ApplicationWaitAfterStopVariableGetter.java

```diff
@@ -0,0 +1,76 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.time.Duration;
+
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
+import org.cloudfoundry.multiapps.controller.process.Constants;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.springframework.util.ObjectUtils;
+
+import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
+
+@Named
+public class ApplicationWaitAfterStopVariableGetter {
+
+    public boolean isAppStopDelayEnvVariableSet(ProcessContext context) {
+        String stopDelayFromEnv = extractDelayVariableFromEnv(context);
+        return !ObjectUtils.isEmpty(stopDelayFromEnv);
+    }
+
+    private String extractDelayVariableFromEnv(ProcessContext context) {
+        CloudApplication appToProcess = context.getVariable(Variables.APP_TO_PROCESS);
+        String waitAfterStopFromAppToProcess = getWaitAfterStopFromApp(appToProcess);
+        if (!ObjectUtils.isEmpty(waitAfterStopFromAppToProcess)) {
+            return waitAfterStopFromAppToProcess;
+        }
+        CloudApplication existingApp = context.getVariable(Variables.EXISTING_APP);
+        return getWaitAfterStopFromApp(existingApp);
+    }
+
+    private String getWaitAfterStopFromApp(CloudApplication app) {
+        if (ObjectUtils.isEmpty(app)) {
+            return null;
+        }
+        return app.getEnv()
+                  .get(Constants.VAR_WAIT_AFTER_APP_STOP);
+    }
```

Comment:
Utilize usage of Optional<> instead of null checks. Return Optional instead of null